CONTRIBUTING.md

```diff
@@ -209,6 +209,72 @@ Before submitting your changes, run the test suite to make sure that nothing is
 ./gradlew check
 ```
 
+### Project layout
+
+This repository is split into many top level directories. The most important
+ones are:
+* docs: Documentation for the project.
+* distribution:  Builds our tar and zip archives and our rpm and deb packages.
+* libs: Libraries used to build other parts of the project. These are meant to
+        be internal rather than general purpose. We have no plans to
+        [semver](https://semver.org/) their APIs or accept feature requests
+        for them. We publish them to maven central because they are
+        dependencies of our plugin test framework, high level rest client, and
+        jdbc driver but.
+* modules:  Features that are shipped with Elasticsearch by default but are not
+        built in to the server. We typically separate features from the server
+        because they require permissions that we don't believe *all* of
+        Elasticsearch should have or because they depend on libraries that we
+        don't believe *all* of Elasticsearch should depend on. For example,
+        reindex requires the `connect` permission so it can perform
+        reindex-from-remote but we don't believe that the *all* of
+        Elasticsearch should have the "connect". For another example, Painless
+        is implemented using antlr4 and asm and we don't believe that *all*
+        of Elasticsearch should have access to them.
+* plugins:  Officially supported plugins to Elasticsearch. We decide that a
```

Comment:
Contains...

Reply:
This is another example of where I'd prefer something other than "we" (and with that in the rest of the list here too). I'm not even sure exactly why, probably because it creates this "we" vs. "you" feeling that I think might sound a bit alienating for external contributors? I might be completely off here, let me know your thoughts.
A way to phrase this differently might be e.g. "A feature is deemed best to be a plugin ... if it seems only important..."

Reply:
Hmmm. I hadn't thought of the "we" vs "you" feeling. I mean, I don't want people to feel excluded. In some sense this is "we, the maintainers" vs "you, the contributor who may or may not want to become a maintainer" and "we" vs "you" is accurate. I'll think about it!

Reply:
I don't like your way because removing the actors makes it seem like the author of the document thinks that the logic is "universal" or "obvious" or something like that. But it isn't. We chose to do the things we chose and it'd have been fairly reasonable to chose otherwise.

Reply:
I understand your point. I will think more about what seems to be bothering me here (reaction to language is interesting I think) but of course we can leave it like it is for this PR.